app/controllers/patients_controller.rb

```diff
@@ -542,6 +548,105 @@ def update_monitoring_fields(patient, params, household, propagation)
     patient.monitoring_history_edit(history_data, diff_state)
   end
 
+  # Update the patient's follow-up flag fields
+  def update_follow_up_flag
+    redirect_to(root_url) && return unless current_user.can_edit_patient? && !current_user.role?(Roles::ENROLLER)
+
+    patient = current_user.get_patient(params.permit(:id)[:id])
```

Comment:
this check already seems to be done already on line 451 so this seems kind of redundant as well